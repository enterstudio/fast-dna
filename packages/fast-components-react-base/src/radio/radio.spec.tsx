import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import {
    IRadioClassNameContract,
    IRadioHandledProps,
    IRadioManagedClasses,
    IRadioUnhandledProps,
    RadioHTMLTags,
    RadioProps
} from "./radio";

/**
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("radio snapshots", (): void => {
    generateSnapshots(examples);
});

describe("radio", (): void => {
    const Component: React.ComponentClass<IRadioHandledProps & IRadioManagedClasses> = examples.component;
    const managedClasses: IRadioClassNameContract = {
        radio: "radio",
        radio_disabled: "radio_disabled",
        radio_input: "radio_input",
        radio_label: "radio_label",
        radio_span: "radio_span"
    };

    const inputSelector: string = `.${managedClasses.radio_input}`;

    test("should implement unhandledProps", () => {
        const handledProps: IRadioHandledProps & IRadioManagedClasses = {
            tag: RadioHTMLTags.div,
            managedClasses
        };

        const unHandledProps: IRadioUnhandledProps = {
            "aria-hidden": true
        };

        const props: RadioProps = {...handledProps, ...unHandledProps};

        const rendered: any = shallow(
            <Component {...props} />
        );

        expect(rendered.first().prop("aria-hidden")).toEqual(true);
    });
});
