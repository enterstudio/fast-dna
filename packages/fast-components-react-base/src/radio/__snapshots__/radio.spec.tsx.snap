// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio snapshots Radio: 0 1`] = `
<div
  className="radio"
>
  <input
    checked={false}
    className="radio_input"
    disabled={null}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="radio_stateIndicator"
  />
</div>
`;

exports[`radio snapshots Radio: 1 1`] = `
<label
  className="radio"
>
  <input
    checked={false}
    className="radio_input"
    disabled={null}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="radio_stateIndicator"
  />
</label>
`;

exports[`radio snapshots Radio: 2 1`] = `
<label
  className="radio"
>
  <input
    checked={true}
    className="radio_input"
    disabled={null}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="radio_stateIndicator"
  />
</label>
`;

exports[`radio snapshots Radio: 3 1`] = `
<label
  className="radio radio_disabled"
>
  <input
    checked={false}
    className="radio_input"
    disabled={true}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="radio_stateIndicator"
  />
</label>
`;
